students/byeongyeon/westagram/users/views.py

```diff
@@ -1,3 +1,35 @@
-from django.shortcuts import render
-
 # Create your views here.
+from django.shortcuts       import render
+import json
+
+from django.http            import JsonResponse
+from django.views           import View
+
+from users.models           import User
+from users.validator        import validate_email, validate_password, validate_mobile
+
+class RegisterView(View):      
+    def post(self, request):
+        data = json.loads(request.body)
+        try:
+            if User.objects.filter(email=data['email']).exists():
+                return JsonResponse({"message": "Email is already in use"}, status=400)
+            elif not validate_email(data["email"]):
+                return JsonResponse({"message": "Email format is invalid"}, status=400)
+            elif not validate_password(data["password"]):
+                return JsonResponse({"message": "Password format is invalid"}, status=400)
+            elif not validate_mobile(data["mobile"]):
+                return JsonResponse({"message": "Mobile number format is invalid"}, status=400)
+            else:         
+                User.objects.create(
+                    name     = data['name'],
+                    email    = data['email'],
+                    password = data['password'],
+                    mobile   = data['mobile'],
+                    address  = data['address'],
+                    age      = data['age']
+                )
```

Comment:
1. if 조건을 만족하게 되면 함수가 종료 되기 때문에 if 아래 로직이 실행 되지 않습니다. 따라서 elif 대신에 if만 가지고도 충분히 구현이 가능하고, 가독성 면에서도 더 좋습니다.

```suggestion
    def post(self, request):
        data = json.loads(request.body)
        
        try:
            if User.objects.filter(email=data['email']).exists():
                return JsonResponse({"message": "Email is already in use"}, status=400)
            
            if not validate_email(data["email"]):
                return JsonResponse({"message": "Email format is invalid"}, status=400)
            
            if not validate_password(data["password"]):
                return JsonResponse({"message": "Password format is invalid"}, status=400)
            
            if not validate_mobile(data["mobile"]):
                return JsonResponse({"message": "Mobile number format is invalid"}, status=400)
                     
            User.objects.create(
                name     = data['name'],
                email    = data['email'],
                password = data['password'],
                mobile   = data['mobile'],
                address  = data['address'],
                age      = data['age']
            )
```